### ISSUE TITLE
Running droned produces an panic

### ISSUE THREAD

**Issue by nooproblem** [drone/drone#1222]
Hi,

I am using a docker container pulled from docker hub, more info here: hub.docker.com/r/drone/drone

I have edited the drone.toml contained in /gopath/src/github.com/drone/drone/packaging/root/etc/drone/ to try to configure drone with a gogs webhook.

However, the following panic message pops up when running "droned --config=/gopath/src/github.com/drone/drone/packaging/root/etc/drone/droned.toml":
```
root@6cedb887917c:/gopath/src/github.com/drone/drone/packaging/root/etc/drone# droned --config=/gopath/src/github.com/drone/drone/packaging/root/etc/drone/drone.toml
panic: listen tcp :80: bind: address already in use

goroutine 1 [running]:
main.main()
	/gopath/src/github.com/drone/drone/server/main.go:157 +0xc37

goroutine 7 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 10 [runnable]:
github.com/drone/drone/server/datastore/database.(*Commitstore).KillCommits(0xc20802b050, 0xc208058500, 0xc20802b050)
	/gopath/src/github.com/drone/drone/server/datastore/database/commit.go:102
created by main.main
	/gopath/src/github.com/drone/drone/server/main.go:105 +0x317

goroutine 9 [chan receive]:
database/sql.(*DB).connectionOpener(0xc208058500)
	/usr/local/go/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:452 +0x31c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
```
The same panic occurs not matter what I do to the drone.toml file (even use the default config), including saving the config file in another location and configuring with an alternative port number.
```
packaging/root/etc/drone# droned
panic: listen tcp :80: bind: address already in use

goroutine 1 [running]:
main.main()
	/gopath/src/github.com/drone/drone/server/main.go:157 +0xc37

goroutine 7 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 9 [chan receive]:
database/sql.(*DB).connectionOpener(0xc208058500)
	/usr/local/go/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:452 +0x31c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 10 [runnable]:
github.com/drone/drone/server/datastore/database.(*Commitstore).KillCommits(0xc2080fd520, 0xc208058500, 0xc2080fd520)
	/gopath/src/github.com/drone/drone/server/datastore/database/commit.go:102
created by main.main
	/gopath/src/github.com/drone/drone/server/main.go:105 +0x317
```
Your thoughts / advice will be greatly appreciated.

**Comment by bradrydzewski** [comment]
two things standout. first is `panic: listen tcp :80: bind: address already in use` seems to indicate that drone cannot bind to port 80 because it is already being used on your server. Also, when running drone in a container, we typically recommend using environment variables instead of toml files, since typically the container won't have access to the toml file on your host machine (unless you choose to mount it in)

**Comment by nooproblem** [comment]
Hi,

Thanks for your reply, it is much appreciated.

As I have stated, if I try configure the toml file to use an alternative port number (say :8080) the same error is produced. Drone is currently bound to port 80:

echo $DRONE_SERVER_PORT 
:80

Additionally, I have tried mounting a data volume to a drone.toml file in the host environment (-v /Users/nooproblem/drone:/gopath/src/github.com/drone/drone/packaging/root/etc/drone) and droned panics just the same.

As you have suggested, I will look into using environmental variables next.

Many thanks!
